Updated index.ts file in the helpers folder of ts-esm-async-await template

Danbaba1 opened the pull request:
**This pull request makes the following changes**
* Fixes (https://discordapp.com/channels/1148391778492354741/1148907665414684692/1227268287704862782)

**General checklist**
- [x] File or folder now contains changes as specified in the issue i worked on
- [ ] I have linked the issue I worked on to this pull request submitted by me

**Testing checklist**

Replace this text with the testing checklist from the issue this pull request fixes.

Ping @code-collabo/node-mongo-api-boilerplate-templates

Danbaba1 commented:
This PR is ready for review.

Ifycode commented on ts-esm-async-await/src/lib/helpers/index.ts:
```diff
@@ -60,8 +60,8 @@ export const npmRunPackageJsonScript = ({ script, currentWorkingDir } : { script
-    const description: string = package_json.description.replace(' generated via Collabo Community\'s create-collabo-app project', '');
-    success(`\nv${package_json.version} ${description}\n\nGenerated via Collabo Community's create-collabo-app project`);
+    const description: string = package_json.description;
+    success(`\nv${package_json.version} ${description}`);
```
The result of what we had before in the terminal:
````
v0.0.0 Typescript (ES Module import/export syntax and async-await asynchronous code handing method) Backend API boilerplate template

Generated via Collabo Community's create-collabo-app project
````

What your own new change result looks like in the terminal:
````
v0.0.0 Typescript (ES Module import/export syntax and async-await asynchronous code handing method) Backend API boilerplate template generated via Collabo Community's create-collabo-app project
````
#

We want it to look like what it was before in the terminal i.e. the spacing and what goes onto a new line. Even though you are to get it directly from the package.json description @Danbaba1

Danbaba1 replied:
Okay noted ma.

Ifycode commented on ts-esm-async-await/src/lib/helpers/index.ts:
```diff
@@ -60,13 +60,14 @@ export const npmRunPackageJsonScript = ({ script, currentWorkingDir } : { script
+    error(`${err}`);
   }
-}
+ }
```
Line 70 @Danbaba1 
The file changes show there's now a space that wasn't there before.
Restore it back, so that it no longer shows up in file changes

Ifycode commented on ts-esm-async-await/src/lib/helpers/index.ts:
```diff
@@ -60,13 +60,14 @@ export const npmRunPackageJsonScript = ({ script, currentWorkingDir } : { script
     success(`\nServer running at ${serverPort}`);
   } catch (err) {
-    error(`${{ err }}`);
+    error(`${err}`);
```
Your line 68 has also changed even though you have no business with this part.
Restore it back, so that it no longer shows up in file changes @Danbaba1

Ifycode commented on ts-esm-async-await/src/lib/helpers/index.ts:
```diff
@@ -60,13 +60,14 @@ export const npmRunPackageJsonScript = ({ script, currentWorkingDir } : { script
 export const server = (serverPort: number | string): void => {
   try {
-    const description: string = package_json.description.replace(' generated via Collabo Community\'s create-collabo-app project', '');
+    const description = package_json.description.split(' generated via')[0];
```
@Danbaba1 
This is what I'm expecting - The success call should look like this (I shouldn't be seeing any part of the string in the success call):
````
success(`\nv${package_json.version} ${description}`);
````

Meanwhile, all the manipulation will happen inside inside `const description` variable. And even when you manipulate, I shouldn't be seeing you writing any part of the string here too, since the string can change.

Danbaba1 commented:
Okay ma.
On Thu, Apr 11, 2024 at 7:47 PM @Ifycode ***@***.***> wrote:

> ***@***.**** requested changes on this pull request.
> ------------------------------
>
> In ts-esm-async-await/src/lib/helpers/index.ts
> <https://github.com/collabo-community/backend-api-boilerplates/pull/8#discussion_r1561469397>
> :
>
> >    }
> -}
> + }
>
> Line 70 @Danbaba1 <https://github.com/Danbaba1>
> The file changes show there's now a space that wasn't there before.
> Restore it back, so that it no longer shows up in file changes
> ------------------------------
>
> In ts-esm-async-await/src/lib/helpers/index.ts
> <https://github.com/collabo-community/backend-api-boilerplates/pull/8#discussion_r1561475343>
> :
>
> >      success(`\nv${package_json.version} ${description}\n\nGenerated via Collabo Community's create-collabo-app project`);
>      success(`\nServer running at ${serverPort}`);
>    } catch (err) {
> -    error(`${{ err }}`);
> +    error(`${err}`);
>
> Your line 68 has also changed even though you have no business with this
> part.
> Restore it back, so that it no longer shows up in file changes @Danbaba1
> <https://github.com/Danbaba1>
> ------------------------------
>
> In ts-esm-async-await/src/lib/helpers/index.ts
> <https://github.com/collabo-community/backend-api-boilerplates/pull/8#discussion_r1561495875>
> :
>
> > @@ -60,13 +60,14 @@ export const npmRunPackageJsonScript = ({ script, currentWorkingDir } : { script
>
>  export const server = (serverPort: number | string): void => {
>    try {
> -    const description: string = package_json.description.replace(' generated via Collabo Community\'s create-collabo-app project', '');
> +    const description = package_json.description.split(' generated via')[0];
>
> @Danbaba1 <https://github.com/Danbaba1>
> This is what I'm expecting - The success call should look like this (I
> shouldn't be seeing any part of the string in the success call):
>
> success(`\nv${package_json.version} ${description}`);
>
> Meanwhile, all the manipulation will happen inside inside const
> description variable. And even when you manipulate, I shouldn't be seeing
> you writing any part of the string here too, since the string can change.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/collabo-community/backend-api-boilerplates/pull/8#pullrequestreview-1994982810>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AXRP57XFBPLP6APCE5Y2PB3Y43LDBAVCNFSM6AAAAABGADYI2WVHI2DSMVQWIX3LMV43YUDVNRWFEZLROVSXG5CSMV3GSZLXHMYTSOJUHE4DEOBRGA>
> .
> You are receiving this because you were mentioned.Message ID:
> <collabo-community/backend-api-boilerplates/pull/8/review/1994982810@
> github.com>
>

Festusali commented:
I believe the changes made matches the requested changes. Hence, the PR can be merged now

Ifycode commented on ts-esm-async-await/package.json:
```diff
@@ -1,7 +1,7 @@
   "name": "ts-esm-async-await-backend-api-boilerplate",
   "version": "0.0.0",
-  "description": "Typescript (ES Module import/export syntax and async-await asynchronous code handing method) Backend API boilerplate template generated via Collabo Community's create-collabo-app project",
+  "description": "Typescript (ES Module import/export syntax and async-await asynchronous code handing method) Backend API boilerplate template \nGenerated via Collabo Community's create-collabo-app project",
```
I disagree @Festusali 
The package.json should be left as it was before, as I have just explained to @Danbaba1

Ifycode replied:
Here's an helpful blog post: https://www.freecodecamp.org/news/javascript-substring-examples-slice-substr-and-substring-methods-in-js

Danbaba1 replied:
Okay ma. Thank you ma.

Danbaba1 commented:
PR is ready for review @collabo-community/review-backend-api

Ifycode commented on ts-esm-async-await/src/lib/helpers/index.ts:
```diff
@@ -58,10 +58,20 @@ export const npmRunPackageJsonScript = ({ script, currentWorkingDir } : { script
+export const capitalizeSecondSentence = (letter: string): string => {
+  letter = package_json.description.slice(126)[0].toUpperCase() + package_json.description.slice(127);
+  return letter;
+}
```
Almost there @Danbaba1 
You tried.
I'll merge this in, but I will share the revamp I did to this instead of allowing you to go on and on on this issue

Ifycode replied:
Also in case of next PR, for consistency, use template literals for strings

Danbaba1 replied:
How do you mean ma?

Ifycode commented on ts-esm-async-await/src/lib/helpers/index.ts:
```diff
@@ -58,10 +58,20 @@ export const npmRunPackageJsonScript = ({ script, currentWorkingDir } : { script
-    const description: string = package_json.description.replace(' generated via Collabo Community\'s create-collabo-app project', '');
-    success(`\nv${package_json.version} ${description}\n\nGenerated via Collabo Community's create-collabo-app project`);
+    const description: string = trimmedDescription(package_json.description) + '\n' + capitalizeSecondSentence(package_json.description);
+    success(`\nv${package_json.version} ${description}`);
```
This is better. But the new line `\n` is supposed to be 2, not 1.
No worries
